### DISCUSSION TITLE
Add Video Duration Filter to Asset Share

### DISCUSSION THREAD

**Discussion by ghost** [Adobe-Marketing-Cloud/asset-share-commons#1198] [General]
Hello 

I would like to suggest adding a filter for video duration to the Asset Share Commons project. Currently, we have the duration data available on the author instance, and it would be highly useful to allow users to filter video assets by their duration directly in the Asset Share search.

This feature would enhance the user experience by enabling more precise searches, particularly when dealing with a large volume of video content.

Please let me know if this is something you would consider implementing, or if I can contribute to this feature in any way.
<img width="497" alt="Capture d’écran 2024-10-31 à 15 45 46" src="https://github.com/user-attachments/assets/8767b29f-5336-404e-8851-5eb6f0390c1f">

thanks

**Comment by davidjgonzalez** [comment]
AEM QueryBuilder has a [range property](https://experienceleague.adobe.com/en/docs/experience-manager-cloud-service/content/implementing/developing/full-stack/search/query-builder-predicates#rangeproperty) that should be able to handle this.

I think the trick here is how users input the lower/upper bounds -- i since the data is stored in seconds, that's what would have to eventually get passed into the query -- but i assume having users enter seconds isn't a great experience -- how are you envisioning user entering these? Seconds? Minutes? Hours + Minutes?

I imagine this would be a "generic" rangeproperty component, that would allow raw numbers to be added, but also special configurations that would handle "time" inputs.

**Reply by ghost** [reply]
I would like it to be in seconds; using a slider might be a good approach. Is this a modification that could be time-consuming?
